otherlibs/stdune/src/appendable_list.ml

```diff
@@ -1,13 +1,38 @@
-type 'a t = 'a list -> 'a list
-
-let empty k = k
-let singleton x k = x :: k
-let to_list l = l []
-let ( @ ) a b k = a (b k)
-let cons x xs = singleton x @ xs
-
-let rec concat l k =
-  match l with
-  | [] -> k
-  | t :: l -> t (concat l k)
+type 'a t =
+  | Empty
+  | Singleton of 'a
+  | Cons of 'a * 'a t
```

Comment:
I wonder if it's also worth having a `List of 'a list` constructor to provide a cheap `of_list` function. Are there any call sites that end up calling ``Cons` a bunch of times to turn a list into an `Appendable_list`?

Reply:
I checked, and it looks like there are no such places. (There are also no calls for `Appendable_list.cons`, so not sure, maybe we could nuke it too... or add `List` anyway.)

Reply:
I added it anyway. There's no cost to it, and it would be bad if we'd get a performance penalty just for using this.

Reply:
Ack, thanks!